library/src/shared/utility/poll/poll.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { discardPeriodicTasks, fakeAsync, tick } from '@angular/core/testing';
+
+import { BehaviorSubject } from 'rxjs';
+import { Poll, PollConfig } from './poll';
+import { Constants } from '@constants';
+
+describe('Poll', () => {
+  let timeout$ = new BehaviorSubject(false);
+  let pollConfig: PollConfig = {
+    timeout: timeout$,
+    interval: Constants.POLL_INTERVAL,
+    duration: Constants.POLL_DURATION
+  };
+  let poll = new Poll(pollConfig);
+
+  it('should start() and return an observable timer based on the set interval', fakeAsync(() => {
+    let count: number = 0;
+
+    poll.start().subscribe((n) => {
+      count = n;
+    });
+    tick(poll.pollConfig.duration);
+
+    // The count should equal the number of times the timer fires in the set duration
+    expect(count).toEqual(5);
```

Comment:
Why 5?

Reply:
Because the poll class need 2 numbers to function properly. The duration of the poll and the interval of the calls. My defaults right now are:

`interval = 1000`
`duration = 5000`

I'm testing the logic here by counting the number of times a `Poll` makes a call. 

I realize these numbers are too small. What do you think would be a reasonable default for these constants?